### DISCUSSION TITLE
Remove PPA from Ubuntu

### DISCUSSION THREAD

**Discussion by kAlvaro** [filips123/PWAsForFirefox#517] [Q&A] [answered]
I'm using Ubuntu 22.04 LTS. I uninstalled the extension a while ago because the app I needed to use actively refuses to be compatible with Firefox. However, I still see `https://packagecloud.io/filips/FirefoxPWA/any` being hit every time I upgrade my system.

I've followed all the documentation I've found about removing PPAs from Ubuntu, to no avail. GUI tools do not show the PPA at all, `sudo add-apt-repository --remove` does not accept any name I type.

There're a couple of files at `/etc/apt/sources.list.d`:

```
-rw-r--r-- 1 root root 114 Apr 24 12:02 firefoxpwa.list
-rw-r--r-- 1 root root 114 Apr 24 12:02 firefoxpwa.list.save
```

Some people say to just delete them, but they contain references to further stuff like `/usr/share/keyrings/firefoxpwa-keyring.gpg`. If my repo settings are already broken, I don't think it's a good idea to break it further.

Do you know how to do a proper clean up?

**Comment by filips123** [comment]
I think you need to remove those two files in `/etc/apt/sources.list.d/` and `/usr/share/keyrings/firefoxpwa-keyring.gpg`.

**Reply by kAlvaro** [reply, marked as answer]
I've finally used the `locate` utility to find for `firefoxpwa` substrings in file names and deleted what showed up. Aside from Trash and home directory, it was this:

```
/etc/apt/sources.list.d/firefoxpwa.list
/etc/apt/sources.list.d/firefoxpwa.list.save
/usr/share/keyrings/firefoxpwa-keyring.gpg
/var/lib/apt/lists/packagecloud.io_filips_FirefoxPWA_any_dists_any_InRelease
/var/lib/apt/lists/packagecloud.io_filips_FirefoxPWA_any_dists_any_main_binary-amd64_Packages
/var/lib/apt/lists/packagecloud.io_filips_FirefoxPWA_any_dists_any_main_binary-i386_Packages
```

If there's something else left, it can't be more broken that it already was :)
